### ISSUE TITLE
`flash_range_program` requires the actual memory address while `flash_range_erase` the address in flash

### ISSUE THREAD

**Issue by TKFRvisionOfficial** [jannic/rp2040-flash#13]
In this crate, `flash_range_module` requires the actual memory address of the rp2040 for the offset while in `flash_range_erase` both `from` and `to` have to be in the form of flash addresses (byte 0 is the first byte of the flash). This is undocumented behavior.

### Example

```rust
use core::ptr;
use rp2040_flash::flash::{flash_range_erase, flash_range_program};

const PICO_FLASH_SIZE_BYTES: u32 = 2097152;
const FLASH_SECTOR_SIZE: u32 = 4096;
const FLASH_PAGE_SIZE: u32 = 256;
const XIP_BASE: u32 = 0x10000000;

const LAST_SECTOR_ADDRESS: u32 = PICO_FLASH_SIZE_BYTES - FLASH_SECTOR_SIZE;
const XIP_LAST_SECTOR_ADDRESS: u32 = XIP_BASE + LAST_SECTOR_ADDRESS;

let to_write = [69u8; FLASH_PAGE_SIZE];

cortex_m::interrupt::free(|_| unsafe {
    // erase at none xip address
    flash_range_erase(LAST_SECTOR_ADDRESS, FLASH_SECTOR_SIZE, true);
    // write at xip address
    flash_range_program(XIP_LAST_SECTOR_ADDRESS, to_write, true);
});

// check if correct byte has been written using unsafe ptr
let ptr_to_start_address = XIP_LAST_SECTOR_ADDRESS as *mut u8;
let first_byte_read = unsafe {
    ptr::read(ptr_to_start_address);
}
assert_eq!(first_byte_read, 69);

cortex_m::interrupt::free(|_| unsafe {
    // delete pages at none xip based address
    flash_range_erase(LAST_SECTOR_ADDRESS, FLASH_SECTOR_SIZE, true);
});

// check if page has been deleted
let second_byte_read = unsafe {
    ptr::read(ptr_to_start_address);
}

assert_eq!(second_byte_read, 0xFF);
```

**Comment by TKFRvisionOfficial** [comment]
After further investigation it looks like `flash_range_program(XIP_LAST_SECTOR_ADDRESS, to_write, true);`  can be replaced by `flash_range_program(LAST_SECTOR_ADDRESS, to_write, true);`. This means that `flash_range_program(XIP_LAST_SECTOR_ADDRESS, to_write, true);` leads to undefined behavior which in this case works by accident. As @jannic explains:

```
The command and address word sent to the flash chip is addr | cmd << 24;
For this to work properly, addr must only use the lower 24 bits, ie. it must be < 0x1000000.
With XIP_LAST_SECTOR_ADDRESS that's not true. So the command sent to the flash chip has some random bits in cmd set from the high bits of the address.
If that works or not depends on which bits are set. If it only affects bits that are set by the correct cmd anyway, it will work. Otherwise, it depends on how the flash chip decodes the cmd.
```

**Comment by jannic** [comment]
Thanks for bringing this to my attention. While it technically was not a bug in rp2040-flash, it's a footgun that can easily be avoided: I added a check `assert!(addr < 0x1000000);` and released version 0.5.1.
